torch/testing/_creation.py

```diff
@@ -143,6 +146,16 @@ def clamp(a, l, h):
     else:
         raise TypeError(f"The requested dtype '{dtype}' is not supported by torch.testing.make_tensor()."
                         " To request support, file an issue at: https://github.com/pytorch/pytorch/issues")
+    if stride_permutation is not None:
+        strides = [None for _ in range(len(shape))]
+        ndim = len(shape)
+        if ndim != len(stride_permutation):
+            raise ValueError(f"Length of shape must equal length of stride permutation, {ndim} != {len(stride_permutation)}")
+        prod = 1
+        for i in range(ndim):
+            strides[stride_permutation[ndim - 1 - i]] = prod  # type: ignore[call-overload]
+            prod *= shape[stride_permutation[ndim - 1 - i]]
```

Comment:
should you check there are no `None` strides at this point? (would happen with invalid stride_permutation)